src/TimeModeling/judiWeights.jl

```diff
@@ -283,4 +290,14 @@ end
 
 function isapprox(x::judiWeights, y::judiWeights; rtol::Real=sqrt(eps()), atol::Real=0)
     isapprox(x.weights, y.weights; rtol=rtol, atol=atol)
-end
\ No newline at end of file
+end
+
+function Base.iterate(x::judiWeights, state=(x.weights[1][1], 1))
```

Comment:
that's a very odd way to iterate and state, do you have a basic example of why this is needed?

Reply:
I got the iterate structure from here: https://julialang.org/blog/2018/07/iterators-in-julia-0.7/
So I needed it so that isLegal(judiWeights) can run which in turn does isnan.(judiWeights)
The iterator is so that any function.() can work.

Reply:
THis would be better with a standard iterate (without indexing the weights) and a proper broadcast overload such as `isnan.` calls `sinan.(weights)`

Reply:
Hmm Im not sure what you mean by standard iterate. Do you mean that we should do isnan.(w.weights) ??
I was trying to avoid that. In spg there is a line that says isLegal(gradient) where gradient is a judiWeights variable. I wanted to avoid having to add logic to catch the case of that gradient being a judiWeights variable etc etc

Reply:
I mean this introduces a weird way to iterate over it as `iterate(w::judiWeight)` does not return `w[1]` so I wouldn't define iterate like that. The for isnan/isinf you can define the broadacted function
```
broadcast(isnan, w::judiWeight)
```

Reply:
>I got the iterate structure from here: https://julialang.org/blog/2018/07/iterators-in-julia-0.7/
That's quite old wouldn't base stuff on 0.7, can check my PR in `judiComposite` for `iterate`